tests/test_diff_tables.py

```diff
@@ -2,36 +2,110 @@
 import unittest
 
 import preql
+import arrow    # comes with preql
 
 from data_diff.database import connect_to_uri
 from data_diff.diff_tables import TableDiffer, TableSegment
 
 from .common import TEST_MYSQL_CONN_STRING, str_to_checksum
 
-
-class TestDiffTables(unittest.TestCase):
+class TestWithConnection(unittest.TestCase):
     @classmethod
     def setUpClass(cls):
         # Avoid leaking connections that require waiting for the GC, which can
         # cause deadlocks for table-level modifications.
         cls.preql = preql.Preql(TEST_MYSQL_CONN_STRING)
         cls.connection = connect_to_uri(TEST_MYSQL_CONN_STRING)
 
+class TestDates(TestWithConnection):
+    def setUp(self):
+        self.connection.query("DROP TABLE IF EXISTS a", None)
+        self.connection.query("DROP TABLE IF EXISTS b", None)
+        self.preql(r"""
+            table a {
+                datetime: datetime
+                comment: string
+            }
+            commit()
+
+            func add(date, comment) {
+                new a(date, comment)
+            }
+        """)
+        self.now = now = arrow.get(self.preql.now())
+        self.preql.add(now.shift(days=-50), "50 days ago")
```

Comment:
When I try to run your tests, this line errors (and presumably the others will too):

<img width="1642" alt="CleanShot 2022-05-11 at 15 45 02@2x" src="https://user-images.githubusercontent.com/97400/167932890-4e2b992b-8fc7-419b-81b9-33a342f4beb4.png">

Would love if `preql` gave some more useful output here 😅

Reply:
This doesn't look related to `preql` in any way. What is Python.Framework? Never heard of it.